src/backend/impl/src/repositories/proposal_repository.rs

```diff
@@ -52,20 +51,9 @@ impl ProposalRepository for ProposalRepositoryImpl {
         Ok(proposal_id)
     }
 
-    fn update_proposal(
-        &self,
-        proposal_id: &ProposalId,
-        proposal: Proposal,
-    ) -> Result<(), ApiError> {
-        self.get_proposal_by_id(proposal_id).ok_or_else(|| {
```

Comment:
While working on this I noticed that we have some redundant checks in the repository that are already performed in the services. I removed it here and in the user profile repository and also added extra tests to the services to make sure these checks are covered.